From cf9b909433e1385ecf9dd71d62d02cab1a0df3c6 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Mon, 4 May 2020 10:34:03 +0530
Subject: [PATCH 0468/1921] octeontx2-af: Fix cpt_rd_wr_register mailbox

cpt_rd_wr_register may be called by AF consumer
before any LF is attached to it hence call
rvu_get_blkaddr accordingly.

Fixes: 4290f456
("octeontx2-af: Update get_rsrc_map for new blocks")

Change-Id: Idb5a0d06bdf134c597f3fe127f61cbfa2e87345e
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/27815
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Srujana Challa <schalla@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
(cherry picked from commit d5b5b5a294bbfad0404ae98a1b4117ceb96c43d2)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/27841
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
index 0002826e3557..ed9e28405edb 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
@@ -462,7 +462,7 @@ int rvu_mbox_handler_cpt_rd_wr_register(struct rvu *rvu,
 	int blkaddr, num_lfs, offs, lf;
 	struct rvu_block *block;
 
-	blkaddr = rvu_get_blkaddr(rvu, BLKTYPE_CPT, req->hdr.pcifunc);
+	blkaddr = rvu_get_blkaddr(rvu, BLKTYPE_CPT, 0);
 	if (blkaddr < 0)
 		return blkaddr;
 
-- 
2.31.1

